tests/api/test_order.py

```diff
@@ -416,6 +416,21 @@ def test_can_finalize_draft_order_no_order_lines(order):
     assert errors[0].message == 'Could not create order without any products.'
 
 
+def test_can_finalize_draft_order_non_existing_variant(order_with_lines):
+    order = order_with_lines
+    line = order.lines.first()
+    variant = line.variant
+    variant.delete()
+    line.refresh_from_db()
```

Comment:
We probably could tell the ORM that we are only looking for the `variant` field.